examples/medplum-scheduling-demo/src/components/CreateUpdateSlot.tsx

```diff
@@ -0,0 +1,140 @@
+import { Button, Modal, Group, Radio } from '@mantine/core';
+import { createReference, getQuestionnaireAnswers, normalizeErrorString } from '@medplum/core';
+import { Questionnaire, QuestionnaireResponse, Schedule, Slot } from '@medplum/fhirtypes';
+import { Event } from 'react-big-calendar';
+import { QuestionnaireForm, useMedplum } from '@medplum/react';
+import { ScheduleContext } from '../Schedule.context';
+import { useContext } from 'react';
+import { showNotification } from '@mantine/notifications';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useToggle } from '@mantine/hooks';
+
+interface CreateUpdateSlotProps {
+  event: Event | undefined;
+  readonly opened: boolean;
+  readonly handlers: {
+    readonly open: () => void;
+    readonly close: () => void;
+    readonly toggle: () => void;
+  };
+}
+
+export function CreateUpdateSlot(props: CreateUpdateSlotProps): JSX.Element {
+  const { event, opened, handlers } = props;
+  const medplum = useMedplum();
+  const { schedule } = useContext(ScheduleContext);
+  const [availableToggle, toggleAvailable] = useToggle([true, false]);
+
+  const editingSlot: Slot = event?.resource;
```

Comment:
one small change - instead of passing in the slot as an extra prop, it should be attached to the `QuestionnaireResponse.subject`. This field typically indicates what the form response is "about". 

You can set this by passing it as a prop to QUestionnaireForm